yktrace/src/swt.rs

```diff
@@ -17,7 +17,8 @@ struct SWTSirTrace {
 impl SWTSirTrace {
     /// Create a SWTSirTrace from a raw buffer and (element) length.
     ///
-    /// The buffer is converted to a higher level representation and the buffer passed in is freed.
+    /// `buf` is assumed to have been allocated by the low-level software tracer via a call to
```

Comment:
I dislike this wording, because it's vague and non-committal. It should be something clearer like  "`buf` must have been allocated via `malloc`; this function will `free` `buf` when it has finished using it."

Reply:
Like so. 
7b024cf

Reply:
Perfect!